cobbler/modules/serializers/mongodb.py

```diff
@@ -24,40 +24,39 @@
 
 from configparser import ConfigParser
 
-pymongo_loaded = False
+from cobbler import settings
+from cobbler.cexceptions import CX
+
 try:
-    from pymongo import Connection
+    from pymongo import MongoClient
     pymongo_loaded = True
-except:
+except ModuleNotFoundError:
     # FIXME: log message
+    pymongo_loaded = False
     pass
 
-import yaml
-
-from cobbler.cexceptions import CX
-
-mongodb = None
+mongodb: MongoClient
 
 
-def __connect():
+def __connect(configfile="/etc/cobbler/mongodb.conf"):
     """
     Reads the config file for mongodb and then connects to the mongodb.
     """
     cp = ConfigParser()
-    cp.read("/etc/cobbler/mongodb.conf")
+    cp.read(configfile)
 
     host = cp.get("connection", "host")
     port = int(cp.get("connection", "port"))
     # TODO: detect connection error
     global mongodb
     try:
-        mongodb = Connection(host, port)['cobbler']
-    except:
+        mongodb = MongoClient(host, port)['cobbler']
+    except Exception:
```

Comment:
That is very broad. Shouldn't we narrow this down a little bit? Might even just be read permissions or something like that.

Reply:
Done